packages/application-components/src/components/drawer/drawer.tsx

```diff
@@ -0,0 +1,178 @@
+import Spacings from '@commercetools-uikit/spacings';
+import Text from '@commercetools-uikit/text';
+import { designTokens as appKitDesignTokens } from '../../theming';
+import {
+  FormPrimaryButton,
+  FormSecondaryButton,
+} from '../internals/default-form-buttons';
+import messages from '../internals/messages';
+import PageHeaderTitle from '../internals/page-header-title';
+import ModalPage from '../modal-pages/internals/modal-page';
+import { LargeIconWrapper } from '../modal-pages/internals/modal-page-top-bar';
+
+type MessageDescriptor = {
+  id: string;
+  description?: string | object;
+  defaultMessage?: string;
+};
+type Label = string | MessageDescriptor;
+
+type TDrawerSize = 10 | 20 | 30;
+
+type TDrawerProps = {
+  size: TDrawerSize;
+  isOpen: boolean;
+  onClose?: (event: SyntheticEvent) => void;
+  children: ReactNode;
+  topBarColor?: 'neutral' | 'surface';
+  // Header Props
+  title: string;
+  subtitle?: string;
+
+  zIndex?: number;
+  getParentSelector?: () => HTMLElement;
+  shouldDelayOnClose?: boolean;
+  afterOpenStyles?: string | CSSObject;
+  /**
+   * This code is used to configure which Custom Views are available for this page.
+   */
+  customViewLocatorCode?: string;
+
+  // Controls Props
+  hideControls?: boolean;
+  formControls?: ReactNode;
+  isPrimaryButtonDisabled?: boolean;
+  isSecondaryButtonDisabled?: boolean;
+  dataAttributesPrimaryButton?: { [key: string]: string };
+  dataAttributesSecondaryButton?: { [key: string]: string };
+  labelPrimaryButton?: Label;
+  labelSecondaryButton?: Label;
+  onPrimaryButtonClick: (event: SyntheticEvent) => void;
+  onSecondaryButtonClick: (event: SyntheticEvent) => void;
+  iconLeftSecondaryButton?: ReactElement;
+};
+
+const defaultProps: Pick<
+  TDrawerProps,
+  'size' | 'hideControls' | 'onPrimaryButtonClick' | 'onSecondaryButtonClick'
+> = {
+  size: 10,
+  hideControls: false,
+  onPrimaryButtonClick: () => {},
+  onSecondaryButtonClick: () => {},
+};
+
+const ContentWrapper = styled.div`
+  height: 100%;
+  padding: ${designTokens.spacing50};
+`;
+
+const getModalPageSize = (size: TDrawerSize) => {
+  switch (size) {
+    case 10:
+      return 'small';
+    case 20:
+      return 'large';
+    case 30:
+      return 'xlarge';
+    default:
+      return 'small';
+  }
+};
+
+function Drawer(props: TDrawerProps) {
+  const intl = useIntl();
+  return (
+    <ModalPage
+      isOpen={props.isOpen}
+      hidePathLabel
+      hideTopBar
+      onClose={props.onClose}
+      size={getModalPageSize(props.size)}
+      title={props.title}
+      afterOpenStyles={props.afterOpenStyles}
+      customViewLocatorCode={props.customViewLocatorCode}
+      getParentSelector={props.getParentSelector}
+      shouldDelayOnClose={props.shouldDelayOnClose}
+      topBarColor={props.topBarColor}
+      zIndex={props.zIndex} // should this be allowed to be set since drawers won't be nested inside each other?
```

Comment:
I don't think we needed now but maybe we can keep it just in case the `Drawer` would be used in a use case that needs it we can't imagine now.

Reply:
Actually this comment was removed already, if you go to "Files changes" in Github, you will see that it's not there.